src/governance/timelock/interface.cairo

```diff
@@ -0,0 +1,123 @@
+    fn get_min_delay(self: @TState) -> u64;
+    fn hash_operation(self: @TState, call: Call, predecessor: felt252, salt: felt252) -> felt252;
+    fn hash_operation_batch(
+        self: @TState, calls: Span<Call>, predecessor: felt252, salt: felt252
+    ) -> felt252;
+    fn schedule(ref self: TState, call: Call, predecessor: felt252, salt: felt252, delay: u64);
+    fn schedule_batch(
+        ref self: TState, calls: Span<Call>, predecessor: felt252, salt: felt252, delay: u64
+    );
+    fn cancel(ref self: TState, id: felt252);
+    fn execute(ref self: TState, call: Call, predecessor: felt252, salt: felt252);
+    fn execute_batch(ref self: TState, calls: Span<Call>, predecessor: felt252, salt: felt252);
+    fn update_delay(ref self: TState, new_delay: u64);
+}
+
+#[starknet::interface]
+pub trait TimelockABI<TState> {
+    fn is_operation(self: @TState, id: felt252) -> bool;
+    fn is_operation_pending(self: @TState, id: felt252) -> bool;
+    fn is_operation_ready(self: @TState, id: felt252) -> bool;
+    fn is_operation_done(self: @TState, id: felt252) -> bool;
+    fn get_timestamp(self: @TState, id: felt252) -> u64;
+    fn get_operation_state(self: @TState, id: felt252) -> OperationState;
+    fn get_min_delay(self: @TState) -> u64;
+    fn hash_operation(self: @TState, call: Call, predecessor: felt252, salt: felt252) -> felt252;
+    fn hash_operation_batch(
+        self: @TState, calls: Span<Call>, predecessor: felt252, salt: felt252
+    ) -> felt252;
+    fn schedule(ref self: TState, call: Call, predecessor: felt252, salt: felt252, delay: u64);
+    fn schedule_batch(
+        ref self: TState, calls: Span<Call>, predecessor: felt252, salt: felt252, delay: u64
+    );
+    fn cancel(ref self: TState, id: felt252);
+    fn execute(ref self: TState, call: Call, predecessor: felt252, salt: felt252);
+    fn execute_batch(ref self: TState, calls: Span<Call>, predecessor: felt252, salt: felt252);
+    fn update_delay(ref self: TState, new_delay: u64);
+
+    // ISRC5
+    fn supports_interface(self: @TState, interface_id: felt252) -> bool;
+
+    // IAccessControl
+    fn has_role(self: @TState, role: felt252, account: ContractAddress) -> bool;
+    fn get_role_admin(self: @TState, role: felt252) -> felt252;
+    fn grant_role(ref self: TState, role: felt252, account: ContractAddress);
+    fn revoke_role(ref self: TState, role: felt252, account: ContractAddress);
+    fn renounce_role(ref self: TState, role: felt252, account: ContractAddress);
+
+    // IAccessControlCamel
+    fn hasRole(self: @TState, role: felt252, account: ContractAddress) -> bool;
+    fn getRoleAdmin(self: @TState, role: felt252) -> felt252;
+    fn grantRole(ref self: TState, role: felt252, account: ContractAddress);
+    fn revokeRole(ref self: TState, role: felt252, account: ContractAddress);
+    fn renounceRole(ref self: TState, role: felt252, account: ContractAddress);
+
+    // IERC721Receiver
+    fn on_erc721_received(
```

Comment:
I think we can remove these from the timelock component, since they are separated components, and can be optionally added in the preset if required. Wdyt?

Reply:
It will definitely reduce some of the dependency noise in the implementation if we remove them. My concern, though, is that I'm not sure it's best to make this opt in as I'd expect token receiver support out of the box (especially with erc1155 transfers). Do you disagree and think opt in is better?

Reply:
To me, opt-in is better. I don't see why every Timelock should be a valid token receiver, since usually Timelocks own contracts for delayed execution of admin actions, and this use case doesn't require being able to receive tokens. It is true that a Timelock could be helpful to "scrow" tokens, but that's why opt-in makes sense IMO. Curious about @Amxx thoughts on this.

Reply:
I would tend toward not including it by default but not because I don't think its usefull. In fact I think most (if not all) timelock should have it.

But the thing is, if you put the 721 hook, then you also need to put the 1155 hook (because why not?) ... but then are there any other hooks you should have ? Possibly ...

If you put them, you are removing that responsability from the user, and when something ends up missing that is on you. If you don't put any, and give that responsability to the user (or to the preset), then you separate the concerns better. On one hand you have the timelock logic that should probably not change a lot, on the other you have the hooks, were new stuff might be included more often.

Reply:
Good points, I'm convinced. Thanks for the feedback! Will update accordingly